src/internal_events/aws_cloudwatch_logs.rs

```diff
@@ -28,42 +29,9 @@ impl InternalEvent for AwsCloudwatchLogsMessageSizeError {
             "error_type" => error_type::ENCODER_FAILED,
             "stage" => error_stage::PROCESSING,
         );
-        counter!(
-            "component_discarded_events_total", 1,
-            "error_code" => "message_too_long",
-            "error_type" => error_type::ENCODER_FAILED,
-            "stage" => error_stage::PROCESSING,
-        );
-    }
-}
-
-#[derive(Debug)]
-pub struct AwsCloudwatchLogsEncoderError {
-    pub error: codecs::encoding::Error,
-}
-
-impl InternalEvent for AwsCloudwatchLogsEncoderError {
-    fn emit(self) {
-        let error_code = io_error_code(&std::io::ErrorKind::InvalidData.into());
-        error!(
-            message = "Error when encoding event.",
-            error = %self.error,
-            error_type = error_type::ENCODER_FAILED,
-            stage = error_stage::PROCESSING,
-            error_code = error_code,
-            internal_log_rate_limit = true,
-        );
-        counter!(
-            "component_errors_total", 1,
-            "error_type" => error_type::ENCODER_FAILED,
-            "error_code" => error_code,
-            "stage" => error_stage::PROCESSING,
-        );
-        counter!(
-            "component_discarded_events_total", 1,
-            "error_type" => error_type::ENCODER_FAILED,
-            "error_code" => error_code,
-            "stage" => error_stage::PROCESSING,
-        );
+        emit!(ComponentEventsDropped::<UNINTENTIONAL> {
+            count: 1,
+            reason: "Encoded event is too long.",
```

Comment:
nit, can extract "Encoded event is too long".

```Rust
let reason = "Encoded event is too long";
```

, then use it as `message = reason` and `reason,` respectively.

Reply:
Good nit, that was a last minute change I did to make the events more "normalized" - updating.

Reply:
Resolved in https://github.com/vectordotdev/vector/pull/14466/commits/cad5a581cb0d721731b9b514bb9f7b7799148bd9